gemrb/core/Map.h

```diff
@@ -427,16 +426,15 @@ class GEM_EXPORT Map : public Scriptable {
 	int CountSummons(ieDword flag, ieDword sex);
 	//returns true if an enemy is near P (used in resting/saving)
 	bool AnyEnemyNearPoint(const Point &p);
-	bool GetBlocked(unsigned int x, unsigned int y, unsigned int size) const;
-	unsigned int GetBlocked(unsigned int x, unsigned int y) const;
-	unsigned int GetBlocked(const Point &p) const;
+	bool CheckSearchmapPointFlags(unsigned int px, unsigned int py, unsigned int size, unsigned int flags, bool actorsAreBlocking = true) const;
+	unsigned int GetBlocked(unsigned int x, unsigned int y, bool actorsAreBlocking = true) const;
 	Scriptable *GetScriptableByGlobalID(ieDword objectID);
 	Door *GetDoorByGlobalID(ieDword objectID);
 	Container *GetContainerByGlobalID(ieDword objectID);
 	InfoPoint *GetInfoPointByGlobalID(ieDword objectID);
-	Actor* GetActorByGlobalID(ieDword objectID);
-	Actor* GetActor(const Point &p, int flags);
-	Actor* GetActorInRadius(const Point &p, int flags, unsigned int radius);
+	Actor* GetActorByGlobalID(ieDword objectID) const;
+	Actor* GetActor(const Point &p, int flags, const Movable* checker = NULL) const;
+	Actor* GetActorInRadius(const Point &p, int flags, unsigned int radius) const;
 	std::vector<Actor *> GetAllActorsInRadius(const Point &p, int flags, unsigned int radius, const Scriptable *see = NULL) const;
 	Actor* GetActor(const char* Name, int flags);
 	Actor* GetActor(int i, bool any) const;
```

Comment:
Moving GetActor next to siblings
```suggestion
	Actor* GetActorInRadius(const Point &p, int flags, unsigned int radius) const;
	std::vector<Actor *> GetAllActorsInRadius(const Point &p, int flags, unsigned int radius, const Scriptable *see = NULL) const;
	Actor* GetActor(const char* Name, int flags);
	Actor* GetActor(int i, bool any) const;
	Actor* GetActor(const Point &p, int flags, const Movable* checker = NULL) const;
```